cmd/runtimecfg/node-ip.go

```diff
@@ -91,53 +94,78 @@ func set(cmd *cobra.Command, args []string) error {
 	}
 	log.Infof("Chosen Node IPs: %v", chosenAddresses)
 
-	// Kubelet
-	kubeletOverrideDir := filepath.Dir(kubeletSvcOverridePath)
-	err = os.MkdirAll(kubeletOverrideDir, 0755)
-	if err != nil {
-		return err
-	}
-	log.Infof("Opening Kubelet service override path %s", kubeletSvcOverridePath)
-	kOverride, err := os.Create(kubeletSvcOverridePath)
-	if err != nil {
-		return err
-	}
-	defer kOverride.Close()
-
 	nodeIP := chosenAddresses[0].String()
 	nodeIPs := nodeIP
 	if len(chosenAddresses) > 1 {
 		nodeIPs += "," + chosenAddresses[1].String()
 	}
+	// Kubelet
 	kOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"KUBELET_NODE_IP=%s\" \"KUBELET_NODE_IPS=%s\"\n", nodeIP, nodeIPs)
 	log.Infof("Writing Kubelet service override with content %s", kOverrideContent)
-	_, err = kOverride.WriteString(kOverrideContent)
+	err = writeToFile(kubeletSvcOverridePath, kOverrideContent)
 	if err != nil {
 		return err
 	}
 
 	// CRI-O
-	crioOverrideDir := filepath.Dir(crioSvcOverridePath)
-	err = os.MkdirAll(crioOverrideDir, 0755)
+	cOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"CONTAINER_STREAM_ADDRESS=%s\"\n", nodeIP)
+	log.Infof("Writing CRIO service override with content %s", cOverrideContent)
+	err = writeToFile(crioSvcOverridePath, cOverrideContent)
+	if err != nil {
+		return err
+	}
+
+	// node ip hint for all other services
+	err = writeToFile(nodeIpFile, nodeIP)
+	if err != nil {
+		return err
+	}
+	for _, ip := range chosenAddresses[0:1] {
```

Comment:
Somewhat confusingly, this doesn't include the 0 and 1 items. It excludes the upper index, so it looks like this needs to be chosenAddresses[0:2] (or just [:2], which seems to be equivalent).

Reply:
totally right, thank you